src/GafferScene/LightToCamera.cpp

```diff
@@ -73,11 +72,8 @@ T parameter( InternedString metadataTarget, const IECore::CompoundData *paramete
 	return defaultValue;
 }
 
-// If the attributes contain a light, return a corresponding camera
-// Just returns the first light it finds, which could produce inconsistent results if you
-// have multiple contradictory lights in the same location.  This seems like a case that should
-// be avoided anyway
-IECore::CameraPtr convertLightToCamera( IECore::ConstCompoundObjectPtr attributes )
+// Return the first light shader found in attributes
+void light( const CompoundObject *attributes, const IECore::CompoundData* &shaderParameters, std::string &metadataTarget )
```

Comment:
I had wondered about having this method return the parameters and name (rather than the Shader), but for a different reason. I was wondering if you would want to maintain support for old lights in caches where the light will be an `IECore::Light` rather than an `IECore::Shader`, and this interface would have allowed that. Is that worth adding?

I'm not particularly fussed because this is all internal implementation detail, but I'd probably use the return value for the shader parameters, because I find output arguments a bit harder to follow.